shelley-mc/formal-spec/utxo.tex

```diff
@@ -145,18 +145,17 @@ \subsection{UTxO Transitions}
   \begin{align*}
     & \fun{getCoin} \in \UTxOOut \to \Coin \\
     & \fun{getCoin}~{\var{out}} ~=~\sum_{\mathsf{adaID} \mapsto tkns \in \fun{getValue}~out}
-       (\sum_{q \in \range~{tkns}} \fun{co}~q) \\
-    & \text{sum total of amount of Ada in an output}
-    \nextdef
-    & \fun{utxoAda} \in \UTxO \to \powerset{(\Addr \times \Coin)} \\
-    & \fun{utxoAda}~{\var{utxo}} ~=~\{~(\fun{getAddr}~\var{out},~\fun{getCoin}~{out})
-    ~\vert~ \var{out} \in \range~\var{utxo} ~\} \\
-    & \text{returns the outputs in the UTxO with only the Ada coins} \\
+       \sum_{q \in \range~{tkns}} \fun{co}~q \\
+    & \text{\emph{-- sum total of amount of Ada in an output}}
     \nextdef
     & \fun{ubalance} \in \UTxO \to \Value \\
     & \fun{ubalance} ~ utxo = \sum_{\wcard\mapsto\var{u}\in~\var{utxo}}
     \fun{getValue}~\var{u} \\
     & \text{UTxO balance} \\
+    \nextdef
+    & \fun{valueSize} \in \Value \to \N \\
+    & \fun{valueSize}~\var{v} = k + k' * |\{ (\var{pid}, \var{aid}) : \var{v}~\var{pid}~\var{aid} \neq 0
```

Comment:
pid and aid are of different size, so you need to count pids and  multiply them by the size coefficient 
ie `| dom v | * k_pid`
then count aid's (add them up for all pid's) and multiply them by a different size coefficient 
ie something like `\sum_{pid |-> tkns \in  dom v, aid \in dom tkns} k_aid`

Reply:
While that's probably more accurate, I don't think it's necessary. We only care about this function for preventing an attack that tries to fill memory, so it doesn't necessarily have to be that precise. And I think there is some value in keeping this function simple, as this is easier to explain to users what the minimum deposit is (a fixed amount for ada, a different fixed amount for every non ada token).